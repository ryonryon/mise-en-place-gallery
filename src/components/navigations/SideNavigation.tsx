import styled from "@emotion/styled";
import * as React from "react";

import Logo from "./NavigationLogo";
import Node from "./NavigationNode";
import { white } from "../../constants/colors";

interface Props {
  logo: string;
  navHeight: string;
  navWidth: string;
  items: [string, string[] | null][];
}

const SideNavigation = ({ logo, navHeight, navWidth, items }: Props) => {
  const StyledNav = styled.nav`
    height: ${navHeight};
    width: ${navWidth};
    background-color: #323439;
    color: #e5e5e5;
    overflow: scroll;
    text-align: left;

    div:hover {
      color: ${white};
      box-shadow: 1px 1px 4px black;
    }

    & a {
      text-decoration: none;
      color: ${white};
    }
  `;

  return (
    <StyledNav>
      <Logo logo={logo} />
      {items.map(([item, children], index) => {
        return <Node key={index} item={item} childItems={children} />;
      })}
    </StyledNav>
  );
};

SideNavigation.defaultProps = {
  navHeight: "100vh",
  navWidth: "20%"
};

export default SideNavigation;
