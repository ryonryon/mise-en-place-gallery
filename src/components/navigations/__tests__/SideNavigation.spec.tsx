import * as React from "react";
import * as renderer from "react-test-renderer";

import SideNavigation from "../SideNavigation";

describe("SideNavigation test", () => {
  it("tests SideNavigation render correctly", () => {
    const sideNavigation = renderer.create(
      <SideNavigation
        logo="test"
        items={[
          ["test1", ["test child 1", "test child 2"]],
          ["test2", ["test child 1", "test child 2"]]
        ]}
      />
    );

    expect(sideNavigation).toMatchSnapshot();
  });
});
