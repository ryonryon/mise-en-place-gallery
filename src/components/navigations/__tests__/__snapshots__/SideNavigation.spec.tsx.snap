// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation test tests SideNavigation render correctly 1`] = `
<nav
  className="css-1bk21om"
>
  <a
    href=""
  >
    <div
      className="css-3ihe6y"
    >
      test
    </div>
  </a>
  <a
    href="#test1"
  >
    <div
      className="css-72aj2j"
      onClick={[Function]}
    >
      test1
    </div>
  </a>
  <a
    href="#test child 1"
  >
    <div
      className="css-wqun73"
      onClick={[Function]}
    >
      test child 1
    </div>
  </a>
  <a
    href="#test child 2"
  >
    <div
      className="css-wqun73"
      onClick={[Function]}
    >
      test child 2
    </div>
  </a>
  <a
    href="#test2"
  >
    <div
      className="css-72aj2j"
      onClick={[Function]}
    >
      test2
    </div>
  </a>
  <a
    href="#test child 1"
  >
    <div
      className="css-wqun73"
      onClick={[Function]}
    >
      test child 1
    </div>
  </a>
  <a
    href="#test child 2"
  >
    <div
      className="css-wqun73"
      onClick={[Function]}
    >
      test child 2
    </div>
  </a>
</nav>
`;
